packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -233,6 +235,9 @@ export class VirtualNode extends VirtualNodeBase {
    * @see https://github.com/aws/aws-app-mesh-roadmap/issues/120
    */
   public addListener(listener: VirtualNodeListener) {
+    if (!this.serviceDiscoveryConfig) {
+      throw new Error('If you specify a listener, then you must specify service discovery information.');
+    }
```

Comment:
@skinny85 - I realized this validation needs to be placed just one location and thus did not create separate `validate()` method. Please advise if this is acceptable approach.

Reply:
Looks good, the message needs some tweaking though:

```suggestion
      throw new Error('service discovery information is required for a VirtualNode with a listener');
```